spyder/config/manager.py

```diff
@@ -104,7 +105,32 @@ def register_plugin(self, plugin_class):
                 backup=True,
                 raw_mode=True,
                 remove_obsolete=False,
+                external_plugin=True
             )
+
+            # Recreate external plugin configs to deal with part two
+            # (the shortcut confliftcs) of spyder-ide/spyder#11132
+            spyder_config = self._user_config._configs_map['spyder']
+            if check_version(spyder_config._old_version, '54.0.0', '<'):
+                # Remove all previous .ini files
+                try:
+                    plugin_config.cleanup()
+                except EnvironmentError:
+                    pass
+
+                # Recreate config

```

Comment:
Is not this repeating the logic from above?

Reply:
Yes, but I don't know how else to do it because first I need to create the `plugin_config` object in order to remove the previous config files (with `plugin_config.cleanup()`). And second, I need to create a new instance again, because the code to generate the `ini` files for the first time is in the `__init__` method of `UserConfig`.

I could move that code to a method in order to call it from the already created `plugin_config` instance. That would avoid the ugly repetition of code at this point. Do you agree?

Reply:
Nah, it is find just wondering about it but yes it is ok like this

Reply:
Ok, great! So do you approve this one?

Reply:
Yep